DKMS Module Build Failure on Arch Linux

When I try to install `sysdig` on Arch by running `sudo pacman -S sysdig-dkms`, the following error occurred:

```
...
(10/10) installing sysdig-dkms                                                      [################################################] 100%
:: Running post-transaction hooks...
(1/2) Arming ConditionNeedsUpdate...
(2/2) Install DKMS modules
==> dkms install --no-depmod scap/3.0.1+driver -k 6.2.8-arch1-1
Error! Bad return status for module build on kernel: 6.2.8-arch1-1 (x86_64)
Consult /var/lib/dkms/scap/3.0.1+driver/build/make.log for more information.
==> WARNING: `dkms install --no-depmod scap/3.0.1+driver -k 6.2.8-arch1-1' exited 10
```

`/var/lib/dkms/scap/3.0.1+driver/build/make.log`:

```
DKMS make.log for scap-3.0.1+driver for kernel 6.2.8-arch1-1 (x86_64)
Fri Mar 24 07:43:53 PM CST 2023
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/main.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/dynamic_params_table.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/fillers_table.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/flags_table.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/ppm_events.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/ppm_fillers.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/event_table.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/syscall_table.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/ppm_cputime.o
  CC [M]  /var/lib/dkms/scap/3.0.1+driver/build/tp_table.o
/var/lib/dkms/scap/3.0.1+driver/build/main.c: In function ‘scap_init’:
/var/lib/dkms/scap/3.0.1+driver/build/main.c:2788:30: error: assignment to ‘char * (*)(const struct device *, umode_t *)’ {aka ‘char * (*)(const struct device *, short unsigned int *)’} from incompatible pointer type ‘char * (*)(struct device *, umode_t *)’ {aka ‘char * (*)(struct device *, short unsigned int *)’} [-Werror=incompatible-pointer-types]
 2788 |         g_ppm_class->devnode = ppm_devnode;
      |                              ^
cc1: some warnings being treated as errors
make[1]: *** [scripts/Makefile.build:252: /var/lib/dkms/scap/3.0.1+driver/build/main.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:2021: /var/lib/dkms/scap/3.0.1+driver/build] Error 2
```

--- a/cmake/modules/driver.cmake
+++ b/cmake/modules/driver.cmake
@@ -31,8 +31,8 @@ else()
   # In case you want to test against another driver version (or branch, or commit) just pass the variable -
   # ie., `cmake -DDRIVER_VERSION=dev ..`
   if(NOT DRIVER_VERSION)
-    set(DRIVER_VERSION "4.0.0+driver")
-    set(DRIVER_CHECKSUM "SHA256=0f71a4e4492847ce6ca35fe6f9ecdf682f603c878397e57d7628a0cd60a29aed")
+    set(DRIVER_VERSION "4.0.1+driver")
+    set(DRIVER_CHECKSUM "SHA256=424d647aa6b6fa45b6701c4bbc9293c161ed00a08254d83eea419c0a736f280d")
   endif()
 
   # cd /path/to/build && cmake /path/to/source

